web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputerSetup/DocumentConnectMyComputerSetup.tsx

```diff
@@ -82,9 +84,32 @@ function Information(props: { onSetUpAgentClick(): void }) {
 
 function AgentSetup() {
   const ctx = useAppContext();
+  const { rootClusterUri } = useWorkspaceContext();
 
   const [setUpRolesAttempt, runSetUpRolesAttempt] = useAsync(
-    useCallback(() => wait(1_000), [])
+    useCallback(async () => {
+      let certsReloaded = false;
+
+      try {
+        const response = await ctx.connectMyComputerService.createRole(
+          rootClusterUri
+        );
+        certsReloaded = response.certsReloaded;
+      } catch (error) {
+        if ((error.message as string)?.includes('access denied')) {
```

Comment:
It'd be nice to be able to check the gRPC status code instead of a naive match on the error message like this. Alas, it all gets lost when it crosses the context bridge.